pkg/querier/blocks_store_queryable.go

```diff
@@ -357,6 +359,8 @@ func (q *blocksStoreQuerier) selectSorted(sp *storage.SelectHints, matchers ...*
 		storage.EmptySeriesSet()
 	}
 
+	stats.Series += len(resSeriesSets)
```

Comment:
`len(resSeriesSets)` is not the number of series, but series set. A single set contains all series returned by 1 store-gateway. We may return the number of series back from store-gateway response (because counting it here would mean iterating the entire set and it's a waste of resources), but this wouldn't be an accurate count anyway because (1) doesn't take in account series from ingesters (2) series from store-gateways may be duplicated.

An option may be wrapping the `storage.SeriesSet` returned by `querier.Select()` and counting the number of `Series()` returned while iterating `SeriesSet`, updating the stats in the context once the the end of set (or error) is reached.